packages/atclient/src/atclient.c

```diff
@@ -270,14 +270,136 @@ exit: { return ret; }
 }
 
 int atclient_get_publickey(const atclient *atclient, const atclient_atkey *atkey, char *value, const size_t valuelen,
-                           size_t *valueolen) {
+                           size_t *valueolen, bool bypasscache) {
   int ret = 1;
 
-  // TODO: implement
+  // 1. initialize variables
+  atclient_atstr cmdbuffer;
+  atclient_atstr_init_literal(&cmdbuffer, 4096, "plookup:");
+
+  atclient_atstr atkeystr;
+  atclient_atstr_init(&atkeystr, ATKEY_GENERAL_BUFFER_SIZE);
+
+  const size_t recvlen = 4096;
+  atclient_atbytes recv;
+  atclient_atbytes_init(&recv, recvlen);
+
+  cJSON *root = NULL;
+
+  if(atkey->atkeytype != ATCLIENT_ATKEY_TYPE_PUBLICKEY) {
+    ret = 1;
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atkey->atkeytype != ATKEYTYPE_PUBLIC\n");
+    goto exit;
+  }
+
+  // 2. build plookup: command
+  if(bypasscache) {
+    ret = atclient_atstr_append(&cmdbuffer, "bypassCache:true:");
```

Comment:
Actually it is possible to append the whole command at the end by using `snprintf()`, which avoids [Schlemiel the Painter's Algorithm ](https://en.wikichip.org/wiki/schlemiel_the_painter%27s_algorithm) (at this point, using `atclient_atstr_append` isn't too different from using multiple `strcat`). What do you think about that?

Reply:
This is great piece of advice @realvarx 

Let me know what you think from this commit:
https://github.com/atsign-foundation/at_c/pull/137/commits/23737d4cb224619c3f40678a1faeeeb43d558900

Reply:
Resolving this.. thanks @realvarx